client/state/posts/selectors.js

```diff
@@ -359,6 +360,29 @@ export const isEditedPostDirty = createSelector(
 );
 
 /**
+  * Returns true if the edited post has content
+  * (title, excerpt or content not empty)
+  *
+  * @param  {Object}  state  Global state tree
+  * @param  {Number}  siteId Site ID
+  * @param  {Number}  postId Post ID
+  * @return {Boolean}        Whether the edited post has content or not
+ */
+export const editedPostHasContent = createSelector(
+	( state, siteId, postId ) => {
+		const editedPost = getEditedPost( state, siteId, postId );
+		return (
+			!! editedPost &&
+			(
+				some( [ 'title', 'excerpt' ], ( field ) => editedPost[ field ] && !! editedPost[ field ].trim() ) ||
+				! isPostContentEmpty( editedPost.content )
+			)
+		);
+	},
+	( state ) => [ state.posts.items, state.posts.edits ]
+);
```

Comment:
Well! I did a small performance benchmarking (obviously this is not perfect, but I repeated the operation multiple times to check). So I wrote "abc" on the Editor Title Component and then cleared it and I checked the "Total time" of `editorPostHasContent` and compared with/without memoization. The result is that it takes `0.3ms` if not memoized and `0ms` (Actually it's not even shown on the results and the smallest time is 0ms) for the memoized version. 

So yes, there is a performance improvement, Is it worth it? I would say "yes", and I would say, we should memoize a maximum of our selectors because 0.3ms here and 0.3ms there, it's worth it. Thoughts?